### PR TITLE
Fix listRepo when /packages don't exists

### PR DESCRIPTION
**Pull request by garbas**
A simple bug we encounter with @djacu when using opan-nix with jupyterWith.

### CODE REVIEW

**Comment by djacu** [comment]
Specifically that when an empty list is passed to `extraOpamProjects` in [ocaml kernel in jupyterWith](https://github.com/tweag/jupyterWith/blob/caf28b971fc92378ad26905a1abf47b1f8f56cf2/kernels/available/ocaml/default.nix#L17), it causes a build failure.

File: src/opam.nix [hunk 1]
```diff
@@ -128,9 +128,13 @@ in rec {
     builder ({ inherit src name version; } // importOpam opamFile) resolveEnv;
 
   listRepo = repo:
-    mergeSortVersions (map (p: listToAttrs [ (nameVerToValuePair p) ])
-      (concatMap attrNames
-        (attrValues (readDirRecursive (repo + "/packages")))));
+    if builtins.pathExists (repo + "/packages")
```
Comment (balsoft) [inline src/opam.nix:131]:
```suggestion
    if pathExists (repo + "/packages")
```

File: src/opam.nix [hunk 2]
```diff
@@ -128,9 +128,13 @@ in rec {
...
+      then 
+        mergeSortVersions (map (p: listToAttrs [ (nameVerToValuePair p) ])
+          (concatMap attrNames
+            (attrValues (readDirRecursive (repo + "/packages")))))
+      else
+        {};
```
Comment (balsoft) [inline src/opam.nix]:
```suggestion
    optionalAttrs (pathExists (repo + "/packages"))
        (mergeSortVersions (map (p: listToAttrs [ (nameVerToValuePair p) ])
          (concatMap attrNames
            (attrValues (readDirRecursive (repo + "/packages"))))));
```

Comment (djacu) [reply src/opam.nix]: looks like that should work. I'll test it to be sure.

**Pull request merged**